sources/packages/web/src/components/common/OfferingFormSubmit.vue

```diff
@@ -0,0 +1,247 @@
+        }
+      } catch {
+        setCalculateStudyPeriodBreakdown({} as StudyPeriodBreakdownAPIOutDTO);
+      }
+    };
+
+    const setCalculateStudyPeriodBreakdown = (
+      calculatedValues: StudyPeriodBreakdownAPIOutDTO,
+    ) => {
+      setComponentValue(
+        offeringForm,
+        FORMIO_STUDY_PERIOD_BREAK_DOWN_KEY,
+        calculatedValues,
+      );
+    };
+
+    /**
+     * Move the screen to the first warning banner, is any.
+     */
+    const scrollToWarningBanner = (
+      warnings: ValidationWarningResultAPIOutDTO[],
+    ) => {
+      const [warningToScroll] = warnings;
+      const warningBannerComponent = `formio-component-${warningToScroll.warningType}`;
+      const [warningBanner] = document.getElementsByClassName(
+        warningBannerComponent,
+      );
+      if (warningBanner) {
+        warningBanner.scrollIntoView({ block: "center", behavior: "smooth" });
+      }
+    };
+
+    const validateOffering = async (data: EducationProgramOfferingAPIInDTO) => {
+      try {
+        const validationResult = await validateOfferingData(data);
+        if (validationResult.offeringStatus === OfferingStatus.Approved) {
+          snackBar.success(
+            "Offering was validated successfully and no warnings were found.",
+          );
+        } else if (
+          validationResult.offeringStatus === OfferingStatus.CreationPending
+        ) {
+          scrollToWarningBanner(validationResult.warnings);
+        }
+      } catch {
+        snackBar.error(
+          "Unexpected error happened during the offering validation.",
+        );
+      }
+    };
+
+    const saveOffering = async (data: EducationProgramOfferingAPIInDTO) => {
+      if (props.formMode === OfferingFormModes.AssessmentDataReadonly) {
+        // No validations are needed to update basic offering data.
+        context.emit("submit", data);
+        return;
+      }
+
+      try {
+        const validationResult = await validateOfferingData(data);
+        if (
+          validationResult.offeringStatus === OfferingStatus.CreationPending
+        ) {
+          const proceedWithWarnings =
+            await offeringWarningsModal.value.showModal();
+          if (proceedWithWarnings) {
+            // There are warnings but the user agreed to proceed.
+            context.emit("submit", data);
+          } else {
+            scrollToWarningBanner(validationResult.warnings);
+          }
+        }
+        if (validationResult.offeringStatus === OfferingStatus.Approved) {
+          // The offering is ready to be automatically approved.
+          context.emit("submit", data);
+        }
+      } catch {
+        snackBar.error(
+          "Unexpected error while validating the offering to be saved.",
+        );
+      }
+    };
+
+    const formChanged = (
+      form: FormIOForm<EducationProgramOfferingAPIInDTO>,
+      event: FormIOChangeEvent,
+    ) => {
+      if (event.changed?.component?.tags?.includes("execute-calculation")) {
```

Comment:
This is really good to know. 

But I have a doubt here. When we call the API to perform the calculations, the whole validation of the offering takes place and why are we not following the same for inputs which may result in showing the banner/making the offering pending?

Reply:
It was not in the scope even for the calculations, I did it because the user experience was not so good. I present it to @abschwenker and there is some decision pending about doing it for the banners or just working with some content.
If there is a decision about it I can work on the next PR.

Reply:
As per the last conversation with @abschwenker the contents will be reviewed and the banner will not be automatically updated "on change" for now.